rflx/pyrflx/typevalue.py

```diff
@@ -1126,13 +1130,23 @@ def __update_simplified_mapping(self, field: Optional[Field] = None) -> None:
             if isinstance(v.last, Number):
                 self._simplified_mapping[v.name_last] = v.last
 
+        nxt = self._next_field(INITIAL.name)
+        while nxt:
+            last_field = nxt
+            nxt = self._next_field(last_field)
+            last = self._fields[last_field].last
+            if nxt == FINAL.name or not isinstance(last, Number):
+                break
+            if nxt:
+                continue
+            for l in self._type.outgoing(Field(last_field)):
+                if l.target == FINAL:
+                    self._simplified_mapping[self.__message_last_name] = last
+                    self._simplified_mapping[self.__message_size_name] = last + Number(1)
```

Comment:
This should be more efficient:

```Python
if any(l.target == FINAL for l in self._type.outgoing(Field(last_field))):
    self._simplified_mapping[self.__message_last_name] = last
    self._simplified_mapping[self.__message_size_name] = last + Number(1)
```

Reply:
Fixed in https://github.com/Componolit/RecordFlux/pull/556/commits/352f3bd761512d1957c2e451c95f4b8e8b0a6126.